CheckboxItem/CheckboxItem.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Agate styled item components with a toggleable checkbox.
+ *
+ * @example
+ * <CheckboxItem onToggle={console.log}>
+ * 	Item with a Checkbox
+ * </CheckboxItem>
+ *
+ * @module agate/CheckboxItem
+ * @exports CheckboxItem
+ * @exports CheckboxItemBase
+ */
+
+import kind from '@enact/core/kind';
+import Item from '@enact/ui/Item';
+import {SlotItem as SlotItemBase} from '@enact/ui/SlotItem';
+import {Cell} from '@enact/ui/Layout';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Checkbox from './Checkbox';
+import ToggleItem from './ToggleItem';
+
+import componentCss from './CheckboxItem.module.less';
+
+const SlotItem = kind({
+	name: 'SlotItem',
+	render: (props) => (
+		<SlotItemBase {...props} component={Item} />
+	)
+});
+
+/**
+ * An item with a checkbox component, ready to use in Agate applications.
+ *
+ * `CheckboxItem` may be used to allow the user to select a single option or used as part of a
+ * [Group]{@link ui/Group} when multiple [selections]{@link ui/Group.Group.select} are possible.
+ *
+ * Usage:
+ * ```
+ * <CheckboxItem
+ * 	defaultSelected={selected}
+ * 	onToggle={handleToggle}
+ * >
+ *  Item with a Checkbox
+ * </CheckboxItem>
+ * ```
+ *
+ * @class CheckboxItem
+ * @memberof agate/CheckboxItem
+ * @extends agate/ToggleItem.ToggleItem
+ * @omit iconComponent
+ * @ui
+ * @public
+ */
+const CheckboxItemBase = kind({
+	name: 'CheckboxItem',
+
+	propTypes: /** @lends agate/CheckboxItem.CheckboxItem.prototype */ {
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal Elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `checkboxItem` - The root class name
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		css: PropTypes.object
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'checkboxItem',
+		publicClassNames: ['checkboxItem']
+	},
+
+	computed: {
+		children: ({children}) => <Cell align="center">{children}</Cell>
+	},
+
+	render: (props) => (
+		<ToggleItem
```

Comment:
I'm not sure we need the abstraction of ToggleItem + SlotItem + Item after the changes in #42. `SwitchItem` uses `Item` directly so I'm curious if that's possible here as well. Can you look into that option a bit?

Here's a rough branch to give you an idea of what I was thinking.

https://github.com/enactjs/agate/compare/feature/PLAT-96102...feature/PLAT-96102-ryan?expand=1